### DISCUSSION TITLE
[Question] Backup/restore settings

### DISCUSSION THREAD

**Discussion by tefracky** [valinet/ExplorerPatcher#209] [Q&A] [answered]
Hello together,

first of all thanks a lot for the great software!

As the title says, it is possible to backup/restore the settings easily (in case of updating Windows or to synchronize with another PC)?

**Comment by valinet** [comment, marked as answer]
Questions should be opened in discussion, it clearly states this in capital letters when you enter this section.

Some settings are stored in the registry, in HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher, the rest are stored in the various places they traditionally sat in. Consult the settings.reg file in the repo for more details.

Thanks

**Comment by oroboros** [comment]
Having an easy way to back/restore all those items would be precious, especially since the settings are all over the place. This would also allow users to put everything to their "default" settings, since Win 11 is still changing, and easily revert back to ExplorerPatcher if needed. Please consider.

**Reply by valinet** [reply]
Okay, I have implemented functionality for importing/exporting the configuration data in the latest pre-release (22000.434.41.11). Read more about how it works here: https://github.com/valinet/ExplorerPatcher/wiki/Settings-management

**Comment by oroboros** [comment]
I just tried the "restore default settings", and a few things I noticed that didn't go as expected.

1) The Update "finds" something if you unselect pre-release check. #710 
2) When I went for "Restore Default Settings", I was expecting Windows 11's, not  ExplorerPatcher's (right? because Taskbar 10 was selected after the restore). Actually it might be interesting to see here a) Win11 factory settings as well as b) EP default settings.
3) If I have dark mode enabled on my system, the Properties toolbar preview is black: 
<img width="327" alt="image" src="https://user-images.githubusercontent.com/1668665/150626570-43f44bf7-ea76-4dc4-9b72-6f75bdd284b6.png">

**Comment by valinet** [comment]
Thanks for testing it out.

1. Explained there and in some previous threads as well. Not an issue.
2. Default settings for ExplorerPatcher means Windows 10 taskbar, it is the only default that’s something else than the OS sets because people install ExplorerPatcher mainly for the Windows 10 taskbar. So that’s a default for it. When you first installed EP, it showed the Windows 10 taskbar, right? “Restore defaults” does so in ExplorerPatcher, not Windows, it *just* happens that most of EP’s defaults are the same as Windows’. That’s an EP setting as well, it doesn’t have any meaning for Windows, when you uninstall EP, the Windows 11 taskbar will come back as the single and only default. So it is working as intended.
3. It is explained [here](https://github.com/valinet/ExplorerPatcher/discussions/515), I see that you keep mentioning it. It’s also not an issue, and not even an EP problem.

Other than these, I understand that it worked well, doing its job? Well, that’s great to hear then.

**Reply by oroboros** [reply]
Thanks for taking time to explain! As for bullet point 2, I see where the developer might think that user will understand that the "restore defaults" wouldn't refer to Win11 default settings, but at least this user didn't interpret it this way! Having the option to restore Win11 factory settings as well as EP default settings might be useful to others as well. Just a thought, and thanks for the wonderful work!

**Reply by valinet** [reply]
It’s just the taskbar. There are too many options already, in my opinion. I don’t understand why this is such a big deal, you installed EP and got the Windows 10 taskbar, you restored its settings and still got that, as that was the default from this app’s perspective. Seems pretty logical to me and logical in general when you think from this perspective. 

> Having the option to restore Win11 factory settings

In general? Why would that be EP’s business? Other than that, the only thing it would do different is set the Windows 11 taskbar instead of 10. So, have 2 options for this nuanced difference…?

As I said, the Windows 10 taskbar is considered the default in EP because that is why most people use this software in the first place. I don’t want to even think how many people will be instantly confused when they’d install this and it would seemingly do nothing. Plus, changing defaults now is pretty cumbersome, you have to do a migration path, as otherwise upgrades would change settings for some people, so yeah… idk, for the moment it is like this, I am glad that this functionality works fine and does its job.

**Reply by oroboros** [reply]
Understood. Like I said, I thought maybe you wanted some real-world user feedback, and thanks once again for your time and work on this.

**Reply by valinet** [reply]
Yeah, I was just explaining why some things are the way they are. Also, for some things, maybe I don’t have an idea how to make it work right away, may seem reluctant to it, but long term I keep it there, in my RAM, and eventually might visit it back. This here being an example, some people asked about this before as well. I didn’t have an idea how to pull it off then but yesterday I did, so I stepped on it. Same with the quirk regarding updates. It is a bit irritating because many people mention it again and again and if it’s still like that it’s because I didn’t have an idea right away how to pull it off, and no one else went a step further and gave a shot at actually implementing themselves a mitigation for it, other than merely (re)observing it.

I appreciate the feedback, otherwise I wouldn’t bother replying or just wouldn’t provide the forums here.
